### DISCUSSION TITLE
Custom wrapped daypicker component, Issue with Typescript

### DISCUSSION THREAD

**Discussion by BuhayovA** [gpbl/react-day-picker#2037] [Support] [answered]
I want to create a custom reusable component based on the react-day-picker library. However, I am having issues with typing. When I try to create a wrapper component around the DayPicker from the library and define all the multi-types, I encounter an error of this kind. How can I correctly type this? I will attach screenshots of the my types below.

```typescript
import React from 'react'
import {
  DateRange,
  DayPicker,
  DayPickerMultipleProps,
  DayPickerRangeProps,
  DayPickerSingleProps,
} from 'react-day-picker'

export interface PickDateFieldProps {
  ...
}

export interface DayPickerMultipleProps extends PickDateFieldProps {
  mode: 'multiple'
  selected?: Date[]
  onSelect?: (days?: Date[]) => void
}

export interface DayPickerSingleProps extends PickDateFieldProps {
  mode: 'single'
  selected?: Date
  onSelect?: (day?: Date) => void
}

export interface DayPickerRangeProps extends PickDateFieldProps {
  mode: 'range'
  selected?: DateRange
  onSelect?: (range?: DateRange) => void
}

export type DatePickerProps = DayPickerMultipleProps | DayPickerSingleProps | DayPickerRangeProps

const TestDatePicker: React.FC<DatePickerProps> = ({
  selected,
  mode,
  ...other
}) => {
  // ...
  return (
    <DayPicker
      mode={mode}
      selected={selected}
      {...other}
    />
  )
}

export default TestDatePicker
```

<img width="955" alt="Screenshot 2024-01-24 at 12 18 49 PM" src="https://github.com/gpbl/react-day-picker/assets/71708401/da40b23d-6d5c-4663-b9d3-07df439f1b2f">

**Comment by BuhayovA** [comment]
And of course in future i need to wrapped this component using react-hook-form.

**Comment by gpbl** [comment, marked as answer]
A fix will be available in the next mayor version.
Closing a Duplicate of https://github.com/gpbl/react-day-picker/discussions/1570
